checker/src/main/java/alluxio/checker/HiveIntegrationChecker.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.checker;
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.PrintWriter;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+
+/**
+ * Some Hive queries to test the integration of Hive with Alluxio.
+ * Supports two options: use Alluxio as an option to store Hive tables
+ * and use Alluxio as the Hive default filesystem.
+ *
+ * This checker requires a running Hadoop cluster, but does not require a running Alluxio cluster.
+ * It will check whether Alluxio classes and Alluxio filesystem can be recognized
+ * in Hadoop task nodes.
+ */
+@SuppressFBWarnings("SQL_PREPARED_STATEMENT_GENERATED_FROM_NONCONSTANT_STRING")
+public class HiveIntegrationChecker {
+  private static final Logger LOG = LoggerFactory.getLogger(HiveIntegrationChecker.class);
+  private static final String FAIL_TO_CONNECT_TO_HIVE_MESSAGE = "Unable to connect to Hive, "
+      + "please check your Hive URL, username and password.\n";
+  private static final String FAIL_TO_FIND_CLASS_MESSAGE = "Please distribute "
+      + "the Alluxio client jar on the classpath of the application across different nodes.\n\n"
+      + "Please set HIVE_AUX_JARS_PATH either in shell or conf/hive-env.sh.\n"
+      + "For details, please refer to: "
+      + "https://www.alluxio.org/docs/master/en/Running-Hive-with-Alluxio.html\n";
+  private static final String FAIL_TO_FIND_FS_MESSAGE = "Please check the fs.alluxio.impl property "
+      + "in core-site.xml file of your Hadoop installation.\n\n"
+      + "For details, please refer to: "
+      + "https://www.alluxio.org/docs/master/en/Debugging-Guide.html\n";
+  private static final String TEST_FAILED_MESSAGE = "***** Integration test failed. *****\n";
+  private static final String TEST_PASSED_MESSAGE = "***** Integration test passed. *****\n";
+
+  @Parameter(names = {"-alluxiourl"}, description = "the alluxio cluster url of form "
+      + "alluxio://master_hostname:port")
+  private String mAlluxioURL = "";
+
+  @Parameter(names = {"-mode"}, description = "1 means use Alluxio as one option "
+      + "to store Hive tables, 2 means use Alluxio as the Hive default filesystem")
+  private int mUserMode = 1;
+
+  @Parameter(names = {"-hiveurl"}, description = "a database url "
```

Comment:
`hiveUrl`? also hive is not really in the description, do you mean the hive Url?

Reply:
Maybe here we can support both -hiveUrl and -hiveurl, because it seems that all commandline argument are lower case.

USAGE="Usage: alluxio-checker.sh hive [-mode USER_MODE] [-hiveurl HIVE_URL] [-name HIVE_USER_NAME] [-password HIVE_USER_PASSWORD]